libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1114,13 +1114,34 @@ void AP_GPS::update_primary(void)
     if (primary_instance == GPS_BLENDED_INSTANCE) {
         primary_instance = 0;
         for (uint8_t i=1; i<GPS_MAX_RECEIVERS; i++) {
-            // choose GPS with highest state or higher number of satellites
-            if ((state[i].status > state[primary_instance].status) ||
-                ((state[i].status == state[primary_instance].status) && (state[i].num_sats > state[primary_instance].num_sats))) {
+            // choose GPS with highest state or higher number of
+            // satellites. Reject a GPS with an old update time, as it
+            // may be the old timestamp that triggered the loss of
+            // blending
```

Comment:
So this obviously fixes a bug, so we should probably merge it as-is.

However, it seems now we have two sets of logic for choosing the best of the two GPSs.  I reckon if our "choose best" code which is at the bottom of this function can't choose between the GPS which just caused blending to fail and the good one then *that* really needs fixing....